use-cases/xtclim/src/trainer.py

```diff
@@ -14,56 +14,114 @@
 tracker.start()
 """
 
+from typing import Literal, Optional
 import torch
 import torch.optim as optim
 import numpy as np
 import pandas as pd
+import sys
+import os
+
+from itwinai.components import monitor_exec
+from itwinai.torch.trainer import TorchTrainer
+from itwinai.torch.distributed import (
+    distributed_resources_available,
+    TorchDistributedStrategy,
+    TorchDDPStrategy,
+    HorovodStrategy,
+    DeepSpeedStrategy,
+    NonDistributedStrategy
+)
+from itwinai.loggers import Logger
+from itwinai.torch.config import TrainingConfiguration
 
-from itwinai.components import Trainer, monitor_exec
 import model
 from torch.utils.data import DataLoader
 from torchvision.utils import make_grid
 from engine import train, validate
 from utils import save_reconstructed_images, save_loss_plot, save_ex
-from initialization import device, beta, criterion
+from initialization import beta, criterion, pixel_wise_criterion
 
-class TorchTrainer(Trainer):
+class XTClimTrainer(TorchTrainer):
     def __init__(
             self,
             epochs: int,
             batch_size: int,
-            lr: float
+            lr: float,
+            strategy: Literal["ddp", "deepspeed", "horovod"] = 'ddp',
+            save_best: bool = True,
+            logger: Optional[Logger] = None
     ):
-        super().__init__()
-        self.epochs = epochs
+        super().__init__(
+            epochs=epochs,
+            config={},
+            strategy=strategy,
+            logger=logger
+        )
         self.batch_size = batch_size
+        self.epochs = epochs
         self.lr = lr
+        # Global training configuration
+        self.config = TrainingConfiguration(
+            batch_size=self.batch_size,
```

Comment:
batch size should be retrieved from `self.config`

Reply:
Done